selfdrive/ui/replay/replay.cc

```diff
@@ -63,13 +63,18 @@ void Replay::mergeEvents() {
   const int end_idx = std::min(current_segment + FORWARD_SEGS, log_paths.size());
 
   // merge logs
-  QMultiMap<uint64_t, Event *> *new_events = new QMultiMap<uint64_t, Event *>();
+  std::vector<Event *> *new_events = new std::vector<Event *>();
   std::unordered_map<uint32_t, EncodeIdx> *new_eidx = new std::unordered_map<uint32_t, EncodeIdx>[MAX_CAMERAS];
   for (int i = start_idx; i <= end_idx; ++i) {
     if (auto it = lrs.find(i); it != lrs.end()) {
```

Comment:
here is a historic bug. LogerrReader may be downloading and updating events in another thread at this point. that will cause invalid data to be merged, and in the worst case, the program may crash.

Reply:
fixed in ##22203